### PR TITLE
Add py.typed file to indicate support for type checkers

### PR DESCRIPTION
**Pull request by Andrew-S-Rosen**
As recommended in [pep 0561](https://peps.python.org/pep-0561/), a blank `py.typed` marker should be included when type hints are used so downstream codes can type check with `mypy` and similar tools.

### CODE REVIEW

**Review by awvwgk — CHANGES_REQUESTED** [review]
The file needs to be included in the distribution, please add it to the meson.build file.

**Comment by Andrew-S-Rosen** [comment]
@awvwgk: Thanks for pointing that out. I added it to `pyproject.toml` just now but am not familiar with the `meson.build` format. I will need some assistance on that.

File: python/pyproject.toml [hunk 1]
```diff
@@ -32,3 +32,6 @@ optional-dependencies.test = [
   "pytest",
   "pytest-cov",
 ]
+
+[tool.setuptools.package-data]
```
Comment (awvwgk) [inline python/pyproject.toml]: We don't use setuptools for this project. The package is defined in `python/tblite/mesin.build` and `py.typed` can be added next to use regular Python sources there.

Comment (Andrew-S-Rosen) [reply python/pyproject.toml]: I have removed the lines from `pyproject.toml` but remain unsure how `meson.build` works. I made a change but don't know if I did it right.

**Review by awvwgk — APPROVED** [review]
Thanks for sharing. I added the py.typed file to the list of installed files.

File: python/pyproject.toml (hunk 1 above)
Comment (awvwgk) [reply python/pyproject.toml]: GH mobile makes it hard to edit PRs, finally got to open my laptop for this. Generally, meson is quite transparent about building and installing projects, so there are no implicit rules or automatic discovery steps like in setuptools. While setuptools is the defacto standard for building Python, it makes it incredible hard to build non-Python components. In meson both Python and non-Python have first class support, which makes it easier to handle the compilation at the cost of slightly more verbose installation instructions for the Python part.

**Comment by Andrew-S-Rosen** [comment]
Thank you for your assistance.

**Pull request merged**